src/errors.rs

```diff
@@ -1,19 +1,22 @@
 use libcommon_rs::errors::Error as BaseError;
 use std::error::Error as StdError;
+use std::net::AddrParseError;
 use std::sync::{MutexGuard, PoisonError};
 
 pub type Result<T> = std::result::Result<T, Error>;
 
 #[derive(Debug)]
 pub enum Error {
     Base(BaseError),
-    // Indicating a vector reached max capacity and can not receive new element
+    /// Indicating a vector reached max capacity and can not receive new element
     AtMaxVecCapacity,
     Bincode(bincode::Error),
     Io(std::io::Error),
-    // Indicating read/write operation was unable to read/write complete size of data
+    /// Indicating read/write operation was unable to read/write complete size of data
     Incomplete,
     PoisonError(String),
+    /// Socket address parse error.
+    AddrParse(AddrParseError),
```

Comment:
Why this error enum has been added, I do not se any use of it in the code?

Reply:
Here: https://github.com/Fantom-foundation/libtransport-tcp/blob/0cec040bfe556f33d5d4bb04f3501280f792a9c8/src/lib.rs#L31

Reply:
That is not necessary as bind() would return io::Error if unable to parse address from string.

Reply:
`AddrParseError` is the error returned by the parser. Let's not drop errors when they are given.

Reply:
Bind() function uses https://github.com/rust-lang/rust/blob/09189591c4c4f6784ffd4bbe99eaefbfe1d5e4a4/src/libstd/net/addr.rs#L906 (see https://github.com/rust-lang/rust/blob/c84a7abf8b2c5753179472464dc2baeb86c6fed6/src/libstd/net/tcp.rs#L678 as well) to convert from string to socket address and returns io::Error, so this change is not needed as io::Error is already covered.

Reply:
OK, I can leave the string argument and the `bind` function and remove this error. My idea was that having an address of type `SocketAddr` would have add static correctness. But it's not important.